### ISSUE TITLE
Add support for environment variables

### ISSUE THREAD

**Issue by mdenson** [microsoft/vscode-makefile-tools#680]
I am working on a dynamically linked library project and it would be nice to be able to set LD_LIBRARY_PATH to a local directory prior to running or debugging the test programs. By the way, I have limited understanding of vs code plugins, so perhaps I am just missing something. In my launch.json I have something like this:
`      "environment": [
        {
          "name": "LD_LIBRARY_PATH",
          "value": "../lib"
        }
      ],
`
which works to set the library path prior to executing. When I try to add the same to the settings.json makefile.launchConfigurations it does not set the environment var and the launch fails.

**Comment by gcampbell-msft** [comment]
@mdenson You are correct, scanning our source code and definition for an environment object in the `makefile.launchConfigurations`, we don't currently support an environment modification, I'll mark this as a feature request. 

Thanks! Please upvote the issue as we often use upvotes to prioritize issues.

**Comment by thefasquelle** [comment]
I would also like to be able to modify environment variables when building please. Based on configuration ofc.
